app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,73 @@
+class TripsController < ApplicationController
+  def index
+
+  end
+
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+    if @trip.nil?
+      head :not_found
+    end
+  end
+
+
+  def rating_completed?
+    @trips = Trip.where(id: params[:passenger_id])
+
+    @trips.each do |trip|
+      if trip.rating == nil
+        return false
+      end
+    end
+    return true
+  end
+
+
+  def create
+    if rating_completed?
+      if params[:passenger_id]
+        passenger = Passenger.find_by(id: params[:passenger_id])
```

Comment:
All of the work done in this method could be moved to a model method. An instance method on `Passenger` would probably work well, something like `Passenger#request_trip`.

Note that you can call methods of one model from another model, so `Passenger#request_trip` could call `Driver.available_driver`.